From 04203191c0261c6a12475865c7053e62b79756ee Mon Sep 17 00:00:00 2001
Message-Id: <04203191c0261c6a12475865c7053e62b79756ee@dist-git>
From: Michal Privoznik <mprivozn@redhat.com>
Date: Tue, 7 Mar 2023 15:36:47 +0100
Subject: [PATCH] qemuxml2argvdata: Extend vCPUs placement in
 memory-hotplug-dimm-addr.xml

So far, the memory-hotplug-dimm-addr.xml test case pins its vCPUs
onto CPUs 0-1 which correspond to NUMA node #0 (per
tests/vircaps2xmldata/linux-basic/system/node/node0). Place vCPUs
onto nodes #1 and #2 too so that DIMM <memory/> device can
continue using thread-context after future patches. This
configuration, as-is currently, would make QEMU error out anyway.

Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
Reviewed-by: Kristina Hanicova <khanicov@redhat.com>
Reviewed-by: Andrea Bolognani <abologna@redhat.com>
(cherry picked from commit c4c90063a5955bca9f5afb5fe03502d3503241c3)
Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2185039
Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
---
 tests/qemuxml2argvdata/memory-hotplug-dimm-addr.xml             | 2 +-
 .../memory-hotplug-dimm-addr.x86_64-latest.xml                  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/qemuxml2argvdata/memory-hotplug-dimm-addr.xml b/tests/qemuxml2argvdata/memory-hotplug-dimm-addr.xml
index 92ea679bbe..47486dda0c 100644
--- a/tests/qemuxml2argvdata/memory-hotplug-dimm-addr.xml
+++ b/tests/qemuxml2argvdata/memory-hotplug-dimm-addr.xml
@@ -4,7 +4,7 @@
   <maxMemory slots='16' unit='KiB'>1099511627776</maxMemory>
   <memory unit='KiB'>7434230</memory>
   <currentMemory unit='KiB'>7434230</currentMemory>
-  <vcpu placement='static' cpuset='0-1'>2</vcpu>
+  <vcpu placement='static' cpuset='0-1,4-5,9'>2</vcpu>
   <os>
     <type arch='i686' machine='pc'>hvm</type>
     <boot dev='hd'/>
diff --git a/tests/qemuxml2xmloutdata/memory-hotplug-dimm-addr.x86_64-latest.xml b/tests/qemuxml2xmloutdata/memory-hotplug-dimm-addr.x86_64-latest.xml
index ef671fcfa3..0a32d5491a 100644
--- a/tests/qemuxml2xmloutdata/memory-hotplug-dimm-addr.x86_64-latest.xml
+++ b/tests/qemuxml2xmloutdata/memory-hotplug-dimm-addr.x86_64-latest.xml
@@ -4,7 +4,7 @@
   <maxMemory slots='16' unit='KiB'>1099511627776</maxMemory>
   <memory unit='KiB'>7434230</memory>
   <currentMemory unit='KiB'>7434230</currentMemory>
-  <vcpu placement='static' cpuset='0-1'>2</vcpu>
+  <vcpu placement='static' cpuset='0-1,4-5,9'>2</vcpu>
   <os>
     <type arch='i686' machine='pc'>hvm</type>
     <boot dev='hd'/>
-- 
2.40.0
